sdk/program/src/vote/instruction.rs

```diff
@@ -185,6 +185,20 @@ impl VoteInstruction {
             _ => panic!("Tried to get slot on non simple vote instruction"),
         }
     }
+
+    /// Only to be used on vote instructions (guard with is_simple_vote),  panics otherwise
+    pub fn timestamp(&self) -> Option<UnixTimestamp> {
+        match self {
+            Self::Vote(v) | Self::VoteSwitch(v, _) => v.timestamp,
```

Comment:
the `timestamp` is the tie vote is generated, or vote is received by this validator? My main question is if timestamp is reliable enough to identify refresh vote?

Reply:
the timestamp is from when the vote is generated in `consensus.rs`, see `record_bank_vote` and `maybe_set_timestamp`. The accompanying change is to update the `timestamp` when refreshing votes, see here https://github.com/solana-labs/solana/pull/31908

Reply:
Ah, makes sense. Thanks for explaining.